environment

```diff
@@ -1,9 +1,42 @@
-#!/usr/bin/env bash
+# Resolve the location of this file and set APP_HOME to the root
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ] ; do SOURCE="$(readlink "$SOURCE")"; done
+export APP_HOME="$(cd -P "$(dirname "$SOURCE")" && pwd)"
+
 set -a
-PROJECT_ROOT=`pwd`
-SERVICE_NAME='query'
-DCP_DNS_DOMAIN='data.humancellatlas.org'
-DEPLOYMENT_STAGE="${DEPLOYMENT_STAGE:-dev}"
-AWS_DEFAULT_REGION='us-east-1'
-API_HOST=${SERVICE_NAME}.${DEPLOYMENT_STAGE}.${DCP_DNS_DOMAIN}
+APP_NAME=dcpquery
+STAGE="${DEPLOYMENT_STAGE:-dev}"
+if [[ $STAGE == "prod" ]]; then
+   API_DOMAIN_NAME=query.data.humancellatlas.org
+   API_DNS_ZONE=data.humancellatlas.org.
+else
+   API_DOMAIN_NAME=query.${STAGE}.data.humancellatlas.org
+   API_DNS_ZONE=${STAGE}.data.humancellatlas.org.
+fi
+BUNDLE_EVENTS_QUEUE_NAME=${APP_NAME}-${STAGE}-bundle-events
+ASYNC_QUERIES_QUEUE_NAME=${APP_NAME}-${STAGE}-async-queries
+WEBHOOK_SECRET_NAME=${APP_NAME}/${STAGE}/webhook-auth-config
+
+# Set DEBUG=0 to disable debugging and set the app log level to ERROR.
+# Set DEBUG=1 to change the log level to INFO and cause stack traces to appear in error responses.
+# Set DEBUG=2 to change the log level to DEBUG and cause stack traces to appear in error responses.
+DEBUG=1
+
+TFSTATE_FILE=.terraform/remote.tfstate
+TF_CLI_ARGS_output="--state ${TFSTATE_FILE}"
+TF_CLI_ARGS_init="--backend-config ${APP_HOME}/.terraform/aws_config"
+
+if [[ ! -v AWS_DEFAULT_REGION ]]; then
+   AWS_DEFAULT_REGION=$(aws configure get region)
+fi
+
+AWS_ACCOUNT_ID=$(aws configure get role_arn | cut -f 5 -d : || aws sts get-caller-identity | jq -r .Account)
+TF_S3_BUCKET=tfstate-$AWS_ACCOUNT_ID
+SERVICE_S3_BUCKET="$APP_NAME-$STAGE-$AWS_ACCOUNT_ID"
```

Comment:
do we want to create a new bucket for every environment?

Reply:
I think we should. For example, the prod environment operates in a different AWS account from dev/staging. That means to share a bucket between them, we would have to configure a custom cross-account access policy (not to mention it's generally not seen as a good security practice).

Reply:
but the with the account-id in the name we shouldn't have to worry about cross-account access
Also I thought we had a limited number of buckets?

Reply:
Ah you're right, the account ID makes it easier. Still, you'd need path-specific access policies.

We do have a limited number of buckets, but the limit is high - 100 by default and it can easily be raised upon request.